core/frontend/src/store/commander.ts

```diff
@@ -0,0 +1,185 @@
+
+  public static getInstance(): CommanderStore {
+    if (!CommanderStore.instance) {
+      CommanderStore.instance = new CommanderStore()
+    }
+    return CommanderStore.instance
+  }
+
+  async commandHost(command: string): Promise<void> {
+    return back_axios({
+      method: 'post',
+      url: `${this.API_URL}/command/host`,
+      timeout: 5000,
+      params: {
+        command,
+        i_know_what_i_am_doing: true,
+      },
+    })
+      .then((response) => response.data)
+      .catch((error) => {
+        if (error === backend_offline_error) {
+          return
+        }
+        const message = `Could not send command to host: ${error.response?.data ?? error.message}.`
+        notifier.pushError('COMMANDER_COMMAND_HOST_FAIL', message, true)
+      })
+  }
+
+  async setTime(unixTimeSeconds: number): Promise<void> {
+    return back_axios({
+      method: 'post',
+      url: `${this.API_URL}/set_time`,
+      timeout: 5000,
+      params: {
+        unix_time_seconds: unixTimeSeconds,
+        i_know_what_i_am_doing: true,
+      },
+    })
+      .then((response) => response.data)
+      .catch((error) => {
+        if (error === backend_offline_error) {
+          return
+        }
+        const message = `Could not set time: ${error.response?.data ?? error.message}.`
+        notifier.pushError('COMMANDER_SET_TIME_FAIL', message, true)
+      })
+  }
+
+  async shutdown(shutdownType: ShutdownType): Promise<void> {
+    return back_axios({
+      method: 'post',
+      url: `${this.API_URL}/shutdown`,
+      timeout: 5000,
+      params: {
+        shutdown_type: shutdownType,
+        i_know_what_i_am_doing: true,
+      },
+    })
+      .then((response) => response.data)
+      .catch((error) => {
+        if (error === backend_offline_error) {
+          return
+        }
+
+        // Connection lost/timeout, normal when we are turning off/rebooting
+        if (error.code === 'ECONNABORTED') {
+          return
+        }
+
+        const message = `Could not execute shutdown: ${error.message ?? error.response?.data}.`
+        notifier.pushError('COMMANDER_SHUTDOWN_FAIL', message, true)
+      })
+  }
+
+  async getRaspiCameraLegacy(): Promise<boolean | undefined> {
+    return back_axios({
+      method: 'get',
+      url: `${this.API_URL}/raspi_config/camera_legacy`,
+      timeout: 5000,
+    })
+      .then((response) => response.data?.enabled)
+      .catch((error) => {
+        if (error === backend_offline_error) {
+          return undefined
+        }
+        const message = 'Could not get Raspberry legacy camera configuration:'
+          + ` ${error.response?.data ?? error.message}.`
+        notifier.pushError('COMMANDER_GET_CAMERA_LEGACY_FAIL', message, true)
+        return undefined
+      })
+  }
+
+  async setRaspiCameraLegacy(enable = true): Promise<boolean> {
+    return back_axios({
+      method: 'post',
+      url: `${this.API_URL}/raspi_config/camera_legacy`,
+      timeout: 5000,
```

Comment:
Note: was `3000`

Reply:
The time was updated to make it more robust